src/transferslot.cpp

```diff
@@ -353,6 +353,36 @@ bool chunkmac_map::unserialize(const char*& ptr, const char* end)
     return true;
 }
 
+void chunkmac_map::calcprogress(m_off_t size, m_off_t& chunkpos, m_off_t& progresscompleted, m_off_t* lastblockprogress)
+{
+    chunkpos = 0;
```

Comment:
I don't see where these two variables are set to 0 in transfer.cpp? Did the original code do that?

Reply:
In transfer.cpp, those are passed from the `t = new Transfer` object, and have been initialised to 0 in its constructor.  good to check though, thanks!

Reply:
Isn't that a weird side-effect of the function to set these variables to 0 if the map is empty for instance?

Reply:
Maybe those should be set to 0 by the caller?